TSMarkdownParser/TSMarkdownParser.m

```diff
@@ -146,15 +146,28 @@ + (instancetype)standardParser {
     /* autodetection */
     
     [defaultParser addLinkDetectionWithLinkFormattingBlock:^(NSMutableAttributedString *attributedString, NSRange range, NSString * _Nullable link) {
-        if (!weakParser.skipLinkAttribute) {
-            NSURL *url = [NSURL URLWithString:link] ?: [NSURL URLWithString:
-                                                        [link stringByAddingPercentEscapesUsingEncoding:NSUTF8StringEncoding]];
+        __block BOOL alreadyLinked = NO;
+        [attributedString enumerateAttributesInRange:NSMakeRange(0, attributedString.length)
+                                             options:0
+                                          usingBlock:^(NSDictionary<NSString *,id> * _Nonnull attrs, NSRange existingRange, BOOL * _Nonnull stop) {
+          id value = attrs[NSLinkAttributeName];
+          if (value && NSIntersectionRange(existingRange, range).length != 0) {
+            // this range has a link that overlaps with the autodetect range, so skip
+            alreadyLinked = YES;
+            *stop = YES;
+          }
+        }];
+        if (!alreadyLinked) {
+            if (!weakParser.skipLinkAttribute) {
```

Comment:
If skipLinkAttribute is YES, there is likely no way to have an NSLinkAttributeName already (unless heavy customization of the standardParser, which isn't recommended). So I'll reorder the conditions a bit.